integration-tests/metrics-v4.2.19-expected-changes.patch

```diff
@@ -0,0 +1,6660 @@
++++ b/metrics-benchmarks/src/main/java/com/codahale/metrics/benchmarks/SlidingTimeWindowReservoirsBenchmark.java
+@@ -2,7 +2,6 @@ package com.codahale.metrics.benchmarks;
+ 
+ import com.codahale.metrics.SlidingTimeWindowArrayReservoir;
+ import com.codahale.metrics.SlidingTimeWindowReservoir;
+-import com.codahale.metrics.Snapshot;
+ import java.util.concurrent.TimeUnit;
+ import org.openjdk.jmh.annotations.Benchmark;
+ import org.openjdk.jmh.annotations.Group;
+@@ -42,8 +41,7 @@ public class SlidingTimeWindowReservoirsBenchmark {
+   @Group("slidingTime")
+   @GroupThreads(1)
+   public Object slidingTimeRead() {
+-    Snapshot snapshot = slidingTime.getSnapshot();
+-    return snapshot;
++    return slidingTime.getSnapshot();
+   }
+ 
+   @Benchmark
+@@ -58,8 +56,7 @@ public class SlidingTimeWindowReservoirsBenchmark {
+   @Group("arrTime")
+   @GroupThreads(1)
+   public Object arrTimeRead() {
+-    Snapshot snapshot = arrTime.getSnapshot();
+-    return snapshot;
++    return arrTime.getSnapshot();
+   }
+ 
+   public static void main(String[] args) throws RunnerException {
+--- a/metrics-collectd/src/main/java/com/codahale/metrics/collectd/CollectdReporter.java
++++ b/metrics-collectd/src/main/java/com/codahale/metrics/collectd/CollectdReporter.java
+@@ -4,10 +4,8 @@ import static com.codahale.metrics.MetricAttribute.COUNT;
+ import static com.codahale.metrics.MetricAttribute.M15_RATE;
+ import static com.codahale.metrics.MetricAttribute.M1_RATE;
+ import static com.codahale.metrics.MetricAttribute.M5_RATE;
+-import static com.codahale.metrics.MetricAttribute.MAX;
+ import static com.codahale.metrics.MetricAttribute.MEAN;
+ import static com.codahale.metrics.MetricAttribute.MEAN_RATE;
+-import static com.codahale.metrics.MetricAttribute.MIN;
+ import static com.codahale.metrics.MetricAttribute.P50;
+ import static com.codahale.metrics.MetricAttribute.P75;
+ import static com.codahale.metrics.MetricAttribute.P95;
+@@ -15,6 +13,7 @@ import static com.codahale.metrics.MetricAttribute.P98;
+ import static com.codahale.metrics.MetricAttribute.P99;
+ import static com.codahale.metrics.MetricAttribute.P999;
+ import static com.codahale.metrics.MetricAttribute.STDDEV;
++import static java.util.Collections.emptySet;
+ 
+ import com.codahale.metrics.Clock;
+ import com.codahale.metrics.Counter;
+@@ -29,7 +28,6 @@ import com.codahale.metrics.Snapshot;
+ import com.codahale.metrics.Timer;
+ import java.io.IOException;
+ import java.net.InetAddress;
+-import java.util.Collections;
+ import java.util.Map;
+ import java.util.Set;
+ import java.util.SortedMap;
+@@ -80,7 +78,7 @@ public class CollectdReporter extends ScheduledReporter {
+     private SecurityLevel securityLevel = SecurityLevel.NONE;
+     private String username = "";
+     private String password = "";
+-    private Set<MetricAttribute> disabledMetricAttributes = Collections.emptySet();
++    private Set<MetricAttribute> disabledMetricAttributes = emptySet();
```

Comment:
Interesting, are we not rewriting this to `ImmutableSet.of`?

Reply:
`CollectorMutability` is disabled, because there's a few edge cases iirc. I can supress those edge cases :+1: and re-enable.